server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/routes/JMAPApiRoutes.scala

```diff
@@ -79,8 +83,12 @@ class JMAPApiRoutes (val authenticator: Authenticator,
 
   private def post(httpServerRequest: HttpServerRequest, httpServerResponse: HttpServerResponse): Mono[Void] =
     SMono(authenticator.authenticate(httpServerRequest))
-      .flatMap((mailboxSession: MailboxSession) => this.requestAsJsonStream(httpServerRequest)
-        .flatMap(requestObject => this.process(requestObject, httpServerResponse, mailboxSession)))
+      .flatMap((mailboxSession: MailboxSession) => SFlux.merge(Seq(
+          userProvisioner.provisionUser(mailboxSession),
+          mailboxesProvisioner.createMailboxesIfNeeded(mailboxSession)))
+        .`then`
+        .`then`(this.requestAsJsonStream(httpServerRequest)
```

Comment:
.then.then(xxx) can likely just be .then(xxx)

Reply:
well... In Flux yeah... In SFlux? It's f******g sh*t ! Couldn't do better...